### PR TITLE
Added Mention Notifications to EastBot

### PR DESCRIPTION
**Pull request by moustafaelsisy**
This PR takes care of making sure that messages send through EastBot to other conversations, and containing @mentions, will trigger a notification for the mentioned user

* Extracted the Notification creation logic to a NotificationService

* Used the NotificationService in the ChatServlet and the BotServlet

* Updated the tests accordingly

### CODE REVIEW

File: src/main/java/codeu/controller/ChatServlet.java [hunk 1]
```diff
@@ -184,23 +185,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
-                                 notificationinfo,
-                                 mentionedUser);
-        notificationStore.addNotification(notification);
-      }
-
-      //Log Activity for message creation
-      String activityDescription = String.format("%s sent a message to %s: %s", username, conversationTitle, messageContent);
-      Activity activity = new Activity(UUID.randomUUID(), activityDescription, Instant.now(), ActivityTypeEnum.MESSAGE_ADDED);
-      activityStore.addActivity(activity);
+    notificationService.generateMentionNotification(message, conversation, user);
```
Comment (moustafaelsisy) [inline src/main/java/codeu/controller/ChatServlet.java:188]: this is the only actual line change in this green block; the rest is just indentation

File: src/main/java/codeu/service/ChatService.java [hunk 2]
```diff
@@ -13,26 +13,36 @@
```
Comment (moustafaelsisy) [inline src/main/java/codeu/service/ChatService.java:13]: This service is used by the ChatBot to send messages to users; thats why the notificationService is called from here

File: src/main/java/codeu/service/NotificationService.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     * @param conversation The conversation the message belongs to
+     * @param author The author of the message
+     */
+    public void generateMentionNotification(Message message, Conversation conversation, User author){
+        UUID mentionedUser = notificationStore.getuserMentioned(message.getContent());
+        //Add notification if a user is mentioned
+        if(mentionedUser != null) {
+
+            String notificationinfo = String.format(
+                    "%s mentioned you in <a href=\"/chat/%s\">%s</a>: %s",
```
Comment (moustafaelsisy) [inline src/main/java/codeu/service/NotificationService.java:42]: Your changes are preserved @ivannpe

File: src/test/java/codeu/controller/ChatServletTest.java [hunk 4]
```diff
@@ -35,6 +35,8 @@
 import javax.servlet.http.HttpServletRequest;
```
Comment (moustafaelsisy) [inline src/test/java/codeu/controller/ChatServletTest.java:35]: Tests from here on

**Review by ivannpe — APPROVED** [review]

**Review by blessinga98 — APPROVED** [review]

**Pull request merged**